From b2ce224fc2794a2cc9692e3e8f6d9bc225e9412f Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowi@redhat.com>
Date: Thu, 7 May 2026 16:20:26 -0400
Subject: [PATCH] copr: update detection of ELN

ELN recently switched its os-release to identify closer to Enterprise Linux
than Fedora, since that more accurately reflects its content and
configuration.

https://github.com/fedora-eln/eln/issues/513
---
 dnf5-plugins/copr_plugin/helpers.cpp             |  2 ++
 .../copr_plugin/tests/data/os-release-fedora-eln | 16 +++++++++-------
 2 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/dnf5-plugins/copr_plugin/helpers.cpp b/dnf5-plugins/copr_plugin/helpers.cpp
index 4bc3bdb3..ff92c13f 100644
--- a/dnf5-plugins/copr_plugin/helpers.cpp
+++ b/dnf5-plugins/copr_plugin/helpers.cpp
@@ -6,6 +6,8 @@ namespace dnf5 {
 
 std::vector<std::string> repo_fallbacks(const std::string & name_version) {
     std::smatch match;
+    if (std::regex_match(name_version, match, std::regex("^eln-([0-9]+)$")))
+        return {name_version, "fedora-eln"};
     if (std::regex_match(name_version, match, std::regex("^(rhel|centos|almalinux)-([0-9]+)$")))
         // try rhel-9-x86_64, then epel-9-x86_64
         return {name_version, "epel-" + match.str(2)};
diff --git a/dnf5-plugins/copr_plugin/tests/data/os-release-fedora-eln b/dnf5-plugins/copr_plugin/tests/data/os-release-fedora-eln
index c42abe78..60d34e7b 100644
--- a/dnf5-plugins/copr_plugin/tests/data/os-release-fedora-eln
+++ b/dnf5-plugins/copr_plugin/tests/data/os-release-fedora-eln
@@ -1,13 +1,15 @@
-NAME="Fedora Linux"
-VERSION="39 (Rawhide Prerelease)"
-ID=fedora
-VERSION_ID=39
+NAME="ELN"
+VERSION="11"
+ID=eln
+ID_LIKE="rhel centos fedora"
+VERSION_ID=11
 VERSION_CODENAME=""
 PLATFORM_ID="platform:eln"
-PRETTY_NAME="Fedora ELN"
-ANSI_COLOR="0;38;2;60;110;180"
+PRETTY_NAME="Fedora ELN 11"
+ANSI_COLOR="0;31"
 LOGO=fedora-logo-icon
-CPE_NAME="cpe:/o:fedoraproject:fedora:39"
+CPE_NAME="cpe:/o:fedoraproject:fedora-eln:11"
+DEFAULT_HOSTNAME="eln"
 HOME_URL="https://fedoraproject.org/"
 DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/eln/"
 SUPPORT_URL="https://ask.fedoraproject.org/"
-- 
2.54.0

